File: 프로그래머스/lv1/12928. 약수의 합/약수의 합.go
package main
import (
    "math"
)

func solution(n int) int {
    answer := 0
    max := int(math.Sqrt(float64(n)))
    for i := 1; i <= max; i++ { 
    if n % i == 0 {
        temp := n / i
        if temp != i {
            answer += i + temp 
        }else{
            answer += i
        }
    }
    }
    return answer
}

File: 프로그래머스/lv1/12928. 약수의 합/약수의 합.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int max = (int)Math.sqrt(n);
        for(int i = 1; i <= max; i++){
            if(n % i == 0){
            int temp = n / i;
            if(temp != i){
            answer += i;
            answer += n / i;
            }
            else{
                answer += i;
            }
            }
        }
        return answer;
    }
}